shuup/utils/money.py

```diff
@@ -60,3 +63,30 @@ def unit_matches_with(self, other):
 
     def new(self, value):
         return type(self)(value, currency=self.currency)
+
+    def as_rounded(self, digits=None):
+        """
+        Returns the value rounded to the given `digits` if specified,
+        or to the fetched `decimal_places` from the Currency
+        loaded from db using the `currency` attribute of this instance.
+
+        In case of non-existent currency in the db,
+        the fallback `digits` will always be 0.
+
+        :param digits int:
+          The number of digits to round
+
+        :return A new value rounded to the given digits
+        """
+
+        if digits is None:
+            # the digits info is not cached yet, fetch it
+            if self.currency not in digits_cache:
+                from shuup.core.models._currencies import Currency
+
+                currency = Currency.objects.filter(code=self.currency).first()
+                digits_cache[self.currency] = currency.decimal_places if currency else 0
+
+            return self.new(numbers.bankers_round(self.value, digits_cache[self.currency]))
+        else:
+            return self.new(numbers.bankers_round(self.value, digits))
```

Comment:
I am proposing another way:

in `utils.money.py`:
```python
def digits_to_precisions(self, mapping):
    return {key: decimal.Decimal('0.1') ** digits for (key, digits) in mapping}

class DefaultPrecisionProvider(object):
    """
    This class provides precision for a given currency.

    Subclass this to implement custom logic and set it
    to the Money class.
    """

    # the most used in the world
    default_precision = decimal.Decimal('0.01')

    # map of precisions for currencies
    currency_precisions = digits_to_precisions({
        'USD': 2,
        'EUR': 2,
        # fill this with more..
    })

    # map os precisions for digits
    digits_precisions = {}

    @classmethod
    def get_precision_for_currency(cls, currency):
        return cls.currency_precisions.get(currency, cls.default_precision)

    @classmethod
    def get_precision_for_digits(cls, digits):
        if digits not in cls.digits_precisions:
            cls.digits_precisions[digits] = (decimal.Decimal('0.1') ** digits)
        return cls.digits_precisions.get(digits)

    @classmethod
    def set_currency_precisions(cls, digits):
        """
        Set default precisions for rounding money amounts.

        :param digits: Map from currency code to number of digits
        :type digits: dict[str,int]
        """
        assert isinstance(digits, dict)
        assert all(isinstance(x, int) for x in digits.values())
        cls.currency_precisions = digits_to_precisions(digits)

class Money(numbers.UnittedDecimal):
    precision_provider = DefaultPrecisionProvider
    # ...

    @classmethod
    def set_precision_provider(cls, precision_provider):
        cls.precision_provider = precision_provider

    def as_rounded(self, digits=None, rounding=ROUND_HALF_EVEN):
        if digits is None:
            precision = self.precision_provider.get_currency_precision(self.currency)
        else:
            precision = self.precision_provider.get_precision_for_digits(digits)

        return self.new(self.value.quantize(precision, rounding=rounding))
```

in `core.models.currencies`:
```python
class CurrencyPrecisionProvider(DefaultPrecisionProvider):
    @classmethod
    def get_precision_for_currency(cls, currency):

        if currency not in cls.currency_precisions:
            currency = Currency.objects.filter(code=currency).first()
            precision = (decimal.Decimal('0.1') ** currency.decimal_places) if currency else self.default_precision
            cls.currency_precisions[currency] = precision

        return cls.currency_precisions.get(currency, cls.default_precision)
```

and finally, in `core.__ini__`:
```python
class ShuupCoreAppConfig(AppConfig):
    # ....
    def ready(self):
        # ....
       Money.set_precision_provider(CurrencyPrecisionProvider)
```

This way we can change the way the precisions are calculated and fetched.